qa/L0_analyze/check_results.py

```diff
@@ -0,0 +1,142 @@
+# Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
```

Comment:
Can you please move this file to `qa/common` folder? We can have a generalized version of this file that allows us to check the model analyzer logs and CSV files of any of the tests. It should accept the arguments below:

* --analyzer-log
* --server-metrics-file
* --config-file
* --model-metrics-gpu-file
* --model-metrics-inference-file
* --expected-number-of-row-inference
* --expected-number-of-row-gpu
* --expected-number-of-row-server
* --expected-number-of-column-inference
* --expected-number-of-column-gpu
* --expected-number-of-column-server

Then, each of the tests can call this Python script with different arguments to make sure that the results are correct. Also, it would be good to use https://docs.python.org/3/library/unittest.html for the checks instead of `check_` functions.

If the test requires additional checks, we can use another `py` file that contains additional unit tests.

Reply:
I think the reason `check_` functions are used instead of the usual `unittest` flow is because they require the arguments you described above. Its not idiomatic to the unittest framework to have arguments. However, if we really want to we can turn the arguments into class variables as in [this post](https://stackoverflow.com/questions/11380413/python-unittest-passing-arguments/20702984)

Reply:
I see. So, it looks like it is better to stick with the current structure and not use unittests.

Reply:
There are some other test cases which don't use the `check_log_table_row_column` and `check_csv_table_row_column`, such as the `L0_perf_analyzer`. Do you mean refactor patterns like 
```
        def print_verification_fail(filename):
            print("\n***\n*** Test Output Verification Failed for " + filename +
                  ".\n***")
            with open(self._analyzer_log, "r") as f:
                print(f.read())

        passed = True

        if not check_log_table_row_column(os.path.abspath(
                self._analyzer_log), server_num_columns, server_num_rows,
                                          "Server\\\\ Only:"):
            print_verification_fail(self._analyzer_log)
            passed = False

        if not check_log_table_row_column(
                os.path.abspath(self._analyzer_log), metrics_num_columns,
                metrics_num_rows, "Models\\\\ \\\\(GPU\\\\ Metrics\\\\):"):
            print_verification_fail(self._analyzer_log)
            passed = False

        if not check_csv_table_row_column(
                os.path.abspath(self._model_metrics_inference_file),
                inference_num_columns, metrics_num_rows, output_tag):
            print_verification_fail(self._model_metrics_inference_file)
            passed = False

        return passed
```
into qa/common and only leave behind things like 
```
        return check1(...) and check2(...) and check3(...)
```
in their respective `check_results.py` files?

Reply:
@kthui If the test doesn't use `check_csv_table_row_column`, then I think it can be converted to a unittest. For the tests that do use the `check_csv_table_row_column` or similar the test will we converted to a Python script and it can called like below in the `test.sh`:

```
python3 ../qa/common/check_results.py --analyzer-log <...> ..
if [ $? -ne 0 ]; then
  echo "\n\n\n Test Failed.\n\n\n"
fi
```